src/WebJobs.Script/StorageProvider/AzureStorageProvider.cs

```diff
@@ -49,25 +57,30 @@ public virtual bool TryGetBlobServiceClientFromConnectionString(out BlobServiceC
         /// <param name="client">client to instantiate</param>
         /// <param name="connection">Name of the connection to use</param>
         /// <returns>successful client creation</returns>
-        public virtual bool TryGetBlobServiceClientFromConnection(out BlobServiceClient client, string connection = null)
+        public virtual bool TryGetBlobServiceClientFromConnection(out BlobServiceClient client, string connection = null, IConfiguration configurationOverride = null)
         {
             var connectionToUse = connection ?? ConnectionStringNames.Storage;
-            return _blobServiceClientProvider.TryGet(connectionToUse, out client);
+            return _blobServiceClientProvider.TryGet(connectionToUse, configurationOverride ?? _configuration, out client);
         }
 
-        public virtual BlobContainerClient GetBlobContainerClient()
+        public virtual BlobContainerClient GetBlobContainerClient(IConfiguration configurationOverride = null)
         {
             if (_storageOptions?.CurrentValue.InternalSasBlobContainer != null)
             {
                 return new BlobContainerClient(new Uri(_storageOptions.CurrentValue.InternalSasBlobContainer));
             }
 
-            if (!TryGetBlobServiceClientFromConnection(out BlobServiceClient blobServiceClient, ConnectionStringNames.Storage))
+            if (!TryGetBlobServiceClientFromConnection(out BlobServiceClient blobServiceClient, ConnectionStringNames.Storage, configurationOverride))
             {
                 throw new InvalidOperationException($"Could not create BlobServiceClient to obtain the BlobContainerClient using Connection: {ConnectionStringNames.Storage}");
             }
 
             return blobServiceClient.GetBlobContainerClient(ScriptConstants.AzureWebJobsHostsContainerName);
         }
+
+        private void InitializeServices()
+        {
+            _configuration = _serviceProvider.GetService<IConfiguration>();
```

Comment:
This might benefit from deferral. No change required for now, but wanted to flag.